src/shared/components/palette/document/write-document-field.component.ts

```diff
@@ -136,35 +145,48 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
       });
     } else {
 
-      this.selectedFile = null;
-      this.uploadedDocument.get('document_url').setValue(null);
-      this.uploadedDocument.get('document_binary_url').setValue(null);
-      this.uploadedDocument.get('document_filename').setValue(null);
-
+      this.setComponentToInitialValues();
       if (this.isAMandatoryComponent()) {
         this.displayFileErrors();
       }
     }
   }
-
-  private createDocumentGroup(url?: string, binaryUrl?: string, filename?: string): void {
-    this.uploadedDocument = this.registerControl(new FormGroup({
-      document_url: new FormControl(url || '', Validators.required),
-      document_binary_url: new FormControl(binaryUrl || '', Validators.required),
-      document_filename: new FormControl(filename || '', Validators.required)
-    }));
+  private setComponentToInitialValues() {
```

Comment:
this method name is a bit misleading. According to the implementation, this should be called: `setComponentToInitialValuesIfMandatory`

Reply:
I'd call this `setMandatoryDocumentToEmtpyValues`. That's what the method is doing

Reply:
DONE